test/rules-custom.spec.ts

```diff
@@ -0,0 +1,61 @@
+import 'mocha';
+
+import { DataLayerRule } from '../src/observer';
+import {
+  expectEqual, waitForFS, setupGlobals, ExpectObserver, expectGlobal,
+} from './utils/mocha';
+
+describe('Custom rules', () => {
+  afterEach(() => {
+    ExpectObserver.getInstance().cleanup();
+  });
+
+  it('should only trigger rules for property changes that match rule selector', async () => {
+    const s = {
+      eVar11: 'test',
+      eVar22: 'test',
+    };
+
+    setupGlobals(([
+      ['s', s],
+    ]));
+
+    const rules: DataLayerRule[] = [
+      {
+        source: 's[^(eVar1)]',
+        operators: [
+          {
+            name: 'insert',
+            value: 'eVar1',
+          },
+        ],
+        destination: 'FS.event',
+      },
+      {
+        source: 's[^(eVar2)]',
+        operators: [
+          {
+            name: 'insert',
+            value: 'eVar2',
+          },
+        ],
+        destination: 'FS.event',
+      },
+    ];
+
+    ExpectObserver.getInstance().create({ rules });
+
+    expectGlobal('s').eVar11 = 11;
+
+    // Wait for event dispatch to result in rule destination calls
+    let [eventName, payload] = await waitForFS('event');
+    expectEqual('eVar1', eventName);
+    expectEqual(11, payload.eVar11);
+
+    expectGlobal('s').eVar22 = 22;
+
+    [eventName, payload] = await waitForFS('event');
+    expectEqual('eVar2', eventName);
+    expectEqual(22, payload.eVar22);
```

Comment:
Hmm is it possible to verify that no event is fired with `eventName == eVar1` when `eVar22` changes, and vice versa? Not sure if that's already covered in this test case, or maybe worth adding here?

Reply:
This is a good suggestion. I originally avoided doing this because I couldn't think of a better strategy than waiting for some defined period of time and checking that no further calls were made. I think this strategy is reasonable, though, considering it takes ~10-20s to run the entire test suite.